system-tests/tia-maven/src/test/java/com/teamscale/tia/TiaMavenMultipleJobsTest.java

```diff
@@ -0,0 +1,43 @@
+package com.teamscale.tia;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.File;
+import java.nio.file.Paths;
+
+import org.conqat.lib.commons.filesystem.FileSystemUtils;
+import org.conqat.lib.commons.io.ProcessUtils;
+import org.conqat.lib.commons.system.SystemUtils;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test class to check if multiple maven plugins can be started with dynamic
+ * port allocation.
+ */
+public class TiaMavenMultipleJobsTest {
+
+	/**
+	 * Starts multiple maven processes and checks that the ports are dynamically set
+	 * and the servers are correctly started.
+	 */
+	@Test
+	public void testMavenTia() throws Exception {
+		File workingDirectory = new File("maven-dump-local-project");
+		String executable = "./mvnw";
+		if (SystemUtils.isWindows()) {
+			executable = Paths.get("maven-dump-local-project", "mvnw.cmd").toUri().getPath();
+		}
+		for (int i = 0; i < 3; i++) {
+			new ProcessBuilder(executable, "clean", "verify").directory(workingDirectory).start();
```

Comment:
I'm still sometimes getting the log file not found error below, because `mvn clean` fails. Can we run `mvn clean` only once before the parallel build, which seems to fix the issue, or do we actually need it to run for every build job?
```
ERROR StatusLogger Log4j2 could not find a logging implementation. Please add log4j-core to the classpath. Using SimpleLogger to log to the console...
[INFO] Scanning for projects...
[INFO] 
[INFO] -----------------------< org.example:tia-maven >------------------------
[INFO] Building tia-maven 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ tia-maven ---
[INFO] 
[INFO] --- teamscale-maven-plugin:32.2.0-SNAPSHOT:prepare-tia-unit-test (default) @ tia-maven ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.712 s
[INFO] Finished at: 2023-10-05T10:11:15+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.teamscale:teamscale-maven-plugin:32.2.0-SNAPSHOT:prepare-tia-unit-test (default) on project tia-maven: Writing the logging configuration file for the TIA agent failed. Make sure the path /Users/pete/Projects/teamscale-jacoco-agent/teamscale-jacoco-agent-2/system-tests/tia-maven/maven-dump-local-project/target/tia/logback.xml is writeable. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException


java.io.FileNotFoundException: /Users/pete/Projects/teamscale-jacoco-agent/teamscale-jacoco-agent-2/system-tests/tia-maven/maven-dump-local-project/target/tia/agent.log (No such file or directory)
	at java.io.FileInputStream.open0(Native Method)
	at java.io.FileInputStream.open(FileInputStream.java:195)
	at java.io.FileInputStream.<init>(FileInputStream.java:138)
	at org.conqat.lib.commons.filesystem.FileSystemUtils.readFileBinary(FileSystemUtils.java:819)
	at org.conqat.lib.commons.filesystem.FileSystemUtils.readFile(FileSystemUtils.java:784)
	at org.conqat.lib.commons.filesystem.FileSystemUtils.readFile(FileSystemUtils.java:749)
	at com.teamscale.tia.TiaMavenMultipleJobsTest.testMavenTia(TiaMavenMultipleJobsTest.java:38)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at java.util.ArrayList.forEach(ArrayList.java:1259)
	at java.util.ArrayList.forEach(ArrayList.java:1259)
```